workspaces/src/cargo/mod.rs

```diff
@@ -1,129 +1,41 @@
-            "Failed to build project '{:?}'.\n\
-            Stderr:\n\
-            {}\n\
-            Stdout:\n\
-            {}",
-            project_path,
-            String::from_utf8(output.stderr).map_err(|e| ErrorKind::DataConversion.custom(e))?,
-            String::from_utf8(output.stdout).map_err(|e| ErrorKind::DataConversion.custom(e))?,
-        )))
-    }
-}
 
 /// Builds the cargo project located at `project_path` and returns the generated wasm file contents.
 ///
 /// NOTE: This function does not check whether the resulting wasm file is a valid smart
 /// contract or not.
-pub async fn compile_project(project_path: &str) -> Result<Vec<u8>> {
-    let project_path = fs::canonicalize(project_path).map_err(|e| match e.kind() {
-        io::ErrorKind::NotFound => ErrorKind::Io.message(format!(
+pub async fn compile_project(project_path: &str) -> crate::Result<Vec<u8>> {
+    let project_path = std::fs::canonicalize(project_path).map_err(|e| match e.kind() {
+        std::io::ErrorKind::NotFound => ErrorKind::Io.message(format!(
             "Incorrect file supplied to compile_project('{}')",
             project_path
         )),
         _ => ErrorKind::Io.custom(e),
     })?;
-    let messages = build_cargo_project(project_path).await?;
-
-    // We find the last compiler artifact message which should contain information about the
-    // resulting .wasm file
-    let compile_artifact = messages
-        .iter()
-        .filter_map(|m| match m {
-            cargo_metadata::Message::CompilerArtifact(artifact) => Some(artifact),
-            _ => None,
-        })
-        .last()
-        .ok_or_else(|| {
-            ErrorKind::Io.message(
-                "Cargo failed to produce any compilation artifacts. \
-                 Please check that your project contains a NEAR smart contract.",
-            )
-        })?;
-    // The project could have generated many auxiliary files, we are only interested in .wasm files
-    let wasm_files = compile_artifact
-        .filenames
-        .iter()
-        .filter(|f| f.as_str().ends_with(".wasm"))
-        .collect::<Vec<_>>();
-    match wasm_files.as_slice() {
-        [] => Err(ErrorKind::Io.message(
-            "Compilation resulted in no '.wasm' target files. \
-                 Please check that your project contains a NEAR smart contract.",
-        )),
-        [file] => {
-            let file = file.canonicalize().map_err(|e| ErrorKind::Io.custom(e))?;
-            Ok(tokio::fs::read(file)
-                .await
-                .map_err(|e| ErrorKind::Io.custom(e))?)
-        }
-        _ => Err(ErrorKind::Io.message(format!(
-            "Compilation resulted in more than one '.wasm' target file: {:?}",
-            wasm_files
-        ))),
-    }
+    let cargo_near_build_command = cargo_near::BuildCommand {
+        release: true,
+        embed_abi: true,
+        doc: false,
+        color: cargo_near::ColorPreference::Always,
+        no_abi: true,
```

Comment:
Just for future context, there was no analysis on which flags actually make sense to use, so we just followed the existing setup - a simple release-mode build for wasm target.